sweetest/src/test/java/com/mysugr/sweetest/ProvideTest.kt

```diff
@@ -35,155 +34,128 @@ class ProvideTest {
 
-    @Test
-    fun `when configured with any, provides can be used`() {
-        givenConfiguredWithAny()
-        // This test class uses `TestUserServiceFakeSteps`, which uses `provide {...}` for
-        // `TestUserService`.
-        // Test service is then used in `TestViewModelSteps` to create a real instance of
-        // `TestViewModel` using `provide<TestViewModel>()`.
-        TestClassWithFakeSteps().run {
-            junitBefore()
-            exampleTest()
-        }
-    }
+        class TestClass : BaseJUnitTest(moduleTestingConfiguration) {
+            override fun configure() = super.configure()
+                .provide<TestUserService> { FakeTestUserService() }
 
-    @Test
-    fun `provides works together with mockOnly`() {
-        givenServiceConfiguredAsMockOnly()
-        // This test class uses `TestUserServiceMockSteps`, which uses `requireMock` for
-        // `TestUserService`.
-        // Test service is then used in `TestViewModelSteps` to create a real instance of
-        // `TestViewModel` using `provide<TestViewModel>()`.
-        TestClassWithMockSteps().run {
-            junitBefore()
-            exampleTest()
+            val userService by dependency<TestUserService>()
         }
-    }
 
-    @Test(expected = RuntimeException::class)
-    fun `no dependencies configured leads to exception`() {
-        givenNothingConfigured()
-        TestClassWithFakeSteps().run {
+        TestClass().run {
             junitBefore()
-            exampleTest()
+            userService // access dependency
         }
     }
 
     @Test(expected = RuntimeException::class)
     fun `provide cannot be used twice for the same type`() {
-        givenConfiguredWithAny()
-        TestClassWithDoubleProvide().run {
-            junitBefore()
+        val moduleTestingConfiguration = moduleTestingConfiguration {
+            dependency any of<TestUserService>()
+            dependency any of<TestViewModel>()
         }
-    }
 
-    private fun givenNothingConfigured() {
-        moduleTestingConfiguration = moduleTestingConfiguration { }
-    }
+        class TestClass : BaseJUnitTest(moduleTestingConfiguration) {
+            override fun configure() = super.configure()
+                .provide<TestViewModel> { TestViewModel(mock()) }
+                .provide<TestViewModel>()
 
-    private fun givenConfiguredWithAny() {
-        moduleTestingConfiguration = moduleTestingConfiguration {
-            dependency any of<TestUserService>()
-            dependency any of<TestViewModel>()
+            val userService by dependency<TestUserService>()
+            val viewModel by dependency<TestViewModel>()
         }
-    }
 
-    private fun givenServiceConfiguredAsMockOnly() {
-        moduleTestingConfiguration = moduleTestingConfiguration {
-            dependency mockOnly of<TestUserService>()
-            dependency any of<TestViewModel>()
+        TestClass().run {
+            junitBefore()
```

Comment:
hm, can you check at which point the exception is thrown? i think it should fail happen here. but then the subsequent lines can be removed...

Reply:
i tried it and in fact the test still works when you remove the lines below.

Reply:
did it on my own